src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/actions/select/SelectedTestRunLineMarkerContributor.java

```diff
@@ -0,0 +1,97 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.facebook.buck.intellij.ideabuck.actions.select;
+
+import com.facebook.buck.intellij.ideabuck.file.BuckFileUtil;
+import com.intellij.execution.lineMarker.RunLineMarkerContributor;
+import com.intellij.icons.AllIcons;
+import com.intellij.openapi.util.IconLoader;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiIdentifier;
+import com.intellij.psi.PsiKeyword;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiTypeParameterList;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Class denoting the lines that can create buck test configurations.
+ */
+public class SelectedTestRunLineMarkerContributor extends RunLineMarkerContributor {
+
+  private static final String[] TEST_ANNOTATIONS = {"org.junit.Test", "org.testng.annotations.Test"};
+
+  @Nullable
+  @Override
+  public Info getInfo(PsiElement psiElement) {
+    if (psiElement instanceof PsiKeyword) {
+      PsiElement parent = psiElement.getParent();
+      if (parent instanceof PsiClass) {
+        PsiClass psiClass = (PsiClass) parent;
+        boolean isTestClass;
+        for (PsiMethod method : psiClass.getAllMethods()) {
+          isTestClass = isTestMethod(method);
+          if (isTestClass) {
+            return createInfo(psiElement);
+          }
+        }
+      }
+    }
+    if (psiElement instanceof PsiTypeParameterList) {
+      PsiElement parent = psiElement.getParent();
+      if (parent instanceof PsiMethod) {
+        if (isTestMethod((PsiMethod) parent)) {
+          return createInfo(psiElement);
+        }
+      }
+    }
+    return null;
+  }
+
+  private Info createInfo(PsiElement psiElement) {
+    return new Info(
+        IconLoader.getIcon("/icons/buck_icon.png"),
+        psiElement1 -> "Run Test(s)",
+        new BuckRunSelectedTestFromEditorAction("Run Test(s) with Buck", "Run Test(s) with Buck", AllIcons.RunConfigurations.TestState.Run, psiElement.getParent()),
+        new BuckDebugSelectedTestFromEditorAction("Debug Test(s) with Buck", "Debug Test(s) with Buck", IconLoader.getIcon("/icons/actions/Debug.png"), psiElement.getParent()));
+  }
+
+  /**
+   * Check that a method is annotated @Test.
+   *
+   * @param method a {@link PsiMethod} to check.
+   * @return {@code true} if the method has the junit Test annotation. {@code false} otherwise.
+   */
+  private boolean isTestMethod(PsiMethod method) {
+    VirtualFile buckFile = BuckFileUtil.getBuckFile(method.getContext().getContainingFile().getVirtualFile());
```

Comment:
looks like `method.getContext().getContainingFile()` can return `null`. Can you check for that?